extra-cpp-bindings/src/walletconnect.rs

```diff
@@ -355,4 +327,234 @@ impl WalletconnectClient {
             anyhow::bail!("no client");
         }
     }
+
+    /// build cronos(eth) eip155 transaction
+    pub fn sign_eip155_transaction_blocking(
+        &mut self,
+        userinfo: &crate::ffi::WalletConnectTxEip155,
+        address: [u8; 20],
+    ) -> Result<Vec<u8>> {
+        if self.client.is_none() {
+            anyhow::bail!("no client");
+        }
+
+        let client = self.client.as_ref().expect("get client");
+        let signeraddress = Address::from_slice(&address);
+
+        let tx = Eip1559TransactionRequest::new()
+            .to(NameOrAddress::Address(Address::from_str(&userinfo.to)?))
+            .data(userinfo.data.as_slice().to_vec())
+            .gas(U256::from_dec_str(&userinfo.gas)?)
+            .max_priority_fee_per_gas(U256::from_dec_str(&userinfo.gas_price)?)
+            .max_fee_per_gas(U256::from_dec_str(&userinfo.gas_price)?)
+            .nonce(U256::from_dec_str(&userinfo.nonce)?)
+            .chain_id(userinfo.chainid)
+            .value(U256::from_dec_str(&userinfo.value)?);
+        let sessioninfo = walletconnect_save_client(&mut self.rt, client)?;
+        let newclient = walletconnect_restore_client(&mut self.rt, sessioninfo)?;
```

Comment:
is this done because client is not cloneable? one issue is with this is that restoring will keep opening new WS connections -- perhaps ok for this PR, but if the issue is non-cloneable client, then perhaps just make `Arc<Mutex<...>>` wrapper (or if the current client itself has just Arc's inside, then it's ok to make it cloneable)

Reply:
yes, it's not cloneable.
i'll check to use Arc<Mutex..

Reply:
in JsonRpcClient, using lock in async can affect performance
rust not compiles, when client is arc,mutex
```
future created by async block is not `Send`
```

Reply:
if it needs to hold the lock across await points, just use the Tokio's mutex: https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html it won't block other tasks

Reply:
(or Tokio's RwLock: https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html if some of the tasks don't need write access)

Reply:
ok, i'll try to tokio lock

Reply:
wow, tokio mutex is great, works like a charm, thanks
```
let mut connection= self.connection.lock().await;
```